src/app/components/core/dashboard/dashboard.component.ts

```diff
@@ -179,14 +178,56 @@ export class DashboardComponent implements OnInit {
   /**
    *  Refresh graphs
    */
-  public refreshGraph() {
-    this.getStatistics();
+  public refreshGraph(limit, keyToRefresh) {
+    let updatedValue = '';
+    this.statisticsService.getStatistics().
+      subscribe((data: any[]) => {
+        const getObject = data.filter(value => value['key'] === keyToRefresh);
+        updatedValue = getObject[0]['value'];
+      },
+        error => {
+          if (error.status === 0) {
+            console.log('service down', error);
+          } else {
+            this.alertService.error(error.statusText);
+          }
+        });
+    this.statisticsService.getStatisticsHistory(limit, keyToRefresh).
+      subscribe((data: any[]) => {
+        this.graphsToShow.forEach(key => {
+          if (key.itemName === keyToRefresh) {
+            const labels = [];
+            const record = map(data['statistics'], keyToRefresh);
+            const history_ts = map(data['statistics'], 'history_ts');
+            history_ts.forEach(element => {
+              element = moment(element).format('HH:mm:ss');
+              labels.push(element);
+            });
+            this.graphsToShow.map(statistics => {
+              if (statistics.itemName === keyToRefresh) {
+                statistics.chartValue = this.getChartValues(labels, record, 'rgb(144,238,144)');
+                statistics.chartType = 'line';
+                statistics.value = updatedValue;
+                statistics.limit = limit;
+                return statistics;
```

Comment:
is this return required? when we are mapping only. and if required, then what if  `if` condition fails?